protocol/kafkaapi/common.go

```diff
@@ -0,0 +1,24 @@
+package kafkaapi
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/kafka-tester/internal/field_encoder"
```

Comment:
`protocol` shouldn't be importing anything from `internal`, should be the other way around - that's why we keep `Encoder` in `protocol` and `FieldEncoder` in `internal`. Think of `protocol` as pure Kafka-related code, nothing to do with how we handle logging etc.